CONTRIBUTING.md

```diff
@@ -42,7 +42,95 @@ you have written a test like this, make sure it does not fail often by
 uncommenting the code that sets a random seed in `tests/main.cpp` and running
 your test many times.
 
-Information on how to build and run the tests is in the main README.md file.
+### Install Dependencies
+
+<details open>
+<summary>Linux</summary>
+
+Use your distributions' package manager to install the required dependencies. The commands shown below are for Ubuntu.
+
+```bash
+$ sudo apt-get update
+$ sudo apt-get install liblapack-dev
+$ sudo apt-get install libarmadillo-dev
+```
+</details>
+
+<details open>
+<summary>MacOS</summary>
+
+Use a package manager like [Homebrew](https://brew.sh) to get the necessary dependencies.
+
+````bash
+$ brew install lapack armadillo
+````
+</details>
+
+<details open>
+<summary>Windows</summary>
+
+You can install **ensmallen** directly by using [vcpkg](https://github.com/microsoft/vcpkg).
+
+```
+vcpkg install ensmallen:x64-windows
+```
+</details>
+
+### Build and Test
+
+This section describes how to build and test the **ensmallen** library from source. **ensmallen** uses CMake as its build system.
+
+First, clone the source code from Github and change into the cloned directory. Or alternatively, you can download the latest relese from the [website](http://ensmallen.org) and extract it.
+
+```bash
+$ git clone https://github.com/mlpack/ensmallen
+$ cd ensmallen
+
+# - or -
+
+$ wget http://ensmallen.org/files/ensmallen-latest.tar.gz
+$ tar -xvzpf ensmallen-latest.tar.gz
+$ cd ensmallen-latest
+```
+
+Next, make a build directory and change into that directory.
+
+```bash
+$ mkdir build
+$ cd build
+```
+
+Then, run the cmake command followed by the make command in the build directory. If the cmake command fails, you probably have missing dependencies.
+
+```bash
+$ cmake ..
+# or with -w flag to inhibit all warning messages
+# $ cmake -DCMAKE_CXX_FLAGS="-w" -DCMAKE_C_FLAGS="-w" .. #
+$ make -j4
+```
+
+Now, you can either run all of the test or an individual test case with:
+
+```bash
+$ ./ensmallen_tests
+$ ./ensmallen_tests <test name>
+```
+
+Ensmallen uses [Catch2](https://github.com/catchorg/Catch2) as the unit test framework.
```

Comment:
Maybe that should come before the test run section.

Reply:
Including this line around the introductory information on line 81 or around there might be preferable.

Reply:
Okay.